hil/api.py

```diff
@@ -424,14 +425,14 @@ def _have_attachment(nic, query):
             "Project does not have access to given network.")
 
     if _have_attachment(nic, model.NetworkAttachment.network == network):
-        raise errors.BlockedError(
+        raise errors.AttachedResourceError(
             "The network is already attached to the nic.")
 
     if channel is None:
         channel = allocator.get_default_channel()
 
     if _have_attachment(nic, model.NetworkAttachment.channel == channel):
-        raise errors.BlockedError("The channel is already in use on the nic.")
+        raise errors.PendingActionError("The channel is already in use on the nic.")
```

Comment:
Maybe it's worth distinguishing a third case, `AlreadyInUseError`, which could be used here and also in e.g. `project_connect_node`. `AttachedResourceError` means "you release that resource because things need to be detached first", whereas `AlreadyInUseError` means "You can't acquire this resource because it's already in use." I kinda don't like `BlockedError` because it's rather vague.

Reply:
yeah, I like `AlreadyInUseError`.